test/client/dio_interceptor_test.dart

```diff
@@ -0,0 +1,197 @@
+        clientSecret: testClientSecret,
+        headerName: headerName,
+        signaturePattern: signaturePattern,
+        disableAutoRetryOnClockSkew: false,
+        clockSkew: Duration(milliseconds: clockskewMS),
+      );
+
+      final dioAdapter = DioAdapter(dio: dio);
+      dio.httpClientAdapter = dioAdapter;
+
+      // Create the interceptor
+      final interceptor = RequestsSignatureInterceptor(
+        options,
+        dio,
+        getTime: (request) {
+          return now.millisecondsSinceEpoch ~/ 1000;
+        },
+      );
+
+      // Add the interceptor to Dio
+      dio.interceptors.add(interceptor);
+
+      // Register the mock response
+      dioAdapter.onGet(baseEndpoint, (request) {
+        options.disableAutoRetryOnClockSkew
+            ? request.reply(HttpStatus.unauthorized, emptyDataInjection)
+            : request.reply(HttpStatus.ok, emptyDataInjection);
+      });
+
+      // Act
+      final response = await dio.get(baseEndpoint);
+
+      // Assert
+      expect(response.statusCode, HttpStatus.ok);
+    });
+
+    test('Auto-retry disabled when clock skew option is disabled', () async {
+      final options = RequestsSignatureOptions(
+        clientId: testClientId,
+        clientSecret: testClientSecret,
+        headerName: headerName,
+        signaturePattern: signaturePattern,
+        disableAutoRetryOnClockSkew: true,
+      );
+
+      final dioAdapter = DioAdapter(dio: dio);
+      dio.httpClientAdapter = dioAdapter;
+
+      // Create the interceptor
+      final interceptor =
+          RequestsSignatureInterceptor(options, dio, getTime: (request) {
+        return now.millisecondsSinceEpoch ~/ 1000;
+      });
+
+      // Add the interceptor to Dio
+      dio.interceptors.add(interceptor);
+
+      int requestCount = 0;
+
+      dioAdapter.onGet(baseEndpoint, (request) {
+        requestCount++;
+        request.reply(HttpStatus.ok, emptyDataInjection);
+      });
+
+      await dio.get(baseEndpoint);
+
+      expect(requestCount, 1);
+    });
+
+    test('Check if tolerance is considered into calculations', () async {
+      final options = RequestsSignatureOptions(
+        clientId: testClientId,
+        clientSecret: testClientSecret,
+        headerName: headerName,
+        signaturePattern: signaturePattern,
+        disableAutoRetryOnClockSkew: false,
+        clockSkew: Duration(milliseconds: clockskewMS),
+      );
+
+      final dioAdapter = DioAdapter(dio: dio);
+      dio.httpClientAdapter = dioAdapter;
+
+      final serverDate = now.add(Duration(milliseconds: clockskewMS));
+
+      // Create the interceptor
+      final interceptor =
+          RequestsSignatureInterceptor(options, dio, getTime: (request) {
+        return now.millisecondsSinceEpoch ~/ 1000;
+      });
+
+      // Add the interceptor to Dio
+      dio.interceptors.add(interceptor);
+
+      dioAdapter.onGet(baseEndpoint, (request) {
+        request.reply(HttpStatus.ok, {});
```

Comment:
Again, if your server always return OK, how are you testing the validity of the timestamp of the signature?